src/main/java/org/vietabroader/controller/SheetFreshController.java

```diff
@@ -0,0 +1,101 @@
+package org.vietabroader.controller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vietabroader.model.GlobalState;
+import org.vietabroader.model.VASpreadsheet;
+
+import javax.swing.*;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.HashMap;
+
+public class SheetFreshController implements Controller {
```

Comment:
Use prgIndicator to signal user that something is going on. This is a network operator so it could be very slow and user might think the program freezes. Look into [`SpreadsheetConnectController`](https://github.com/vietabroader/VAQR/blob/master/src/main/java/org/vietabroader/controller/SpreadsheetConnectController.java) for example